build/templates/session.py.mako

```diff
@@ -100,6 +104,24 @@ class AttributeViSession(object):
 
     def __set__(self, obj, value):
         raise TypeError('Attributes of type ViSession are unsupported in Python')
+% for c in config['context_manager']:
+<%
+context_name = 'acquisition' if c['direction'] == 'input' else 'generation'
+enter_function = next(f for f in functions if f['name'].upper() == c['enter'].upper())
+exit_function = next(f for f in functions if f['name'].upper() == c['exit'].upper())
+%>\
+
+
+class ${context_name.title()}(object):
+    def __init__(self, session):
+        self.session = session
+
+    def __enter__(self):
+        self.session.${enter_function['python_name']}()
```

Comment:
Are they? If they are I can hard code them, but I like it being part of the code generation.